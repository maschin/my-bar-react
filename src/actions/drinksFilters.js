export const setTextFilter = (text = '') => ({
    type: 'SET_TEXT_FILTER',
    text
});

export const setCategoryFilter = (category = '') => ({
    type: 'SET_CATEGORY_FILTER',
    category
});

export const setStatusFilter = (status = 'Full') => ({
    type: 'SET_STATUS_FILTER',
    status
});