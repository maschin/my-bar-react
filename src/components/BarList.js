import React from 'react';
import { connect } from 'react-redux';

import DrinkListItem from '../components/DrinkListItem';

export const BarList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-desktop">Drink</div>
            <div className="show-for-desktop">Data</div>
        </div>
        <div className="list-body">
            {
                props.drinks.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>Your bar is totally empty!</span>
                    </div>
                ) : (
                    props.drinks.map((drink) => (
                       <DrinkListItem key={drink.id} {...drink} />
                    ))
                )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        drinks: state.drinks
    };
};

export default connect(mapStateToProps)(BarList);