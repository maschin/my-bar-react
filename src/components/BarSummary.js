import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const BarSummary = ({ bottlesCount }) => {
    const bottlesWord = bottlesCount === 1 ? 'bottle' : 'bottles';

    return(
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    You have total <span>{bottlesCount}</span> {bottlesWord} in your bar.
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Drink</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        drinks: state.drinks,
        bottlesCount: state.drinks.length
    };
};

export default connect(mapStateToProps)(BarSummary);