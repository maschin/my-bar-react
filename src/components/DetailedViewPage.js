import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import {
    openDrink,
    emptyDrink
} from '../actions/drinks';

export class DetailedView extends React.Component {
    onOpen = () => {
        this.props.openDrink(this.props.drink);
    }
    onMakeEmpty = () => {
        this.props.emptyDrink(this.props.drink);
    }
    render(){
        return(
            <div>
                {
                    this.props.drink ? (
                        <div>
                            <div className="page-header">
                                <div className="content-container">
                                    <div className="page-header__actions">
                                        <Link className="button" to={`/edit/${this.props.drink.id}`}>Edit</Link>
                                        { !this.props.drink.isOpen ? (
                                            <button id="openDrink" className="button" onClick={this.onOpen}>Open</button>
                                        ) : (

                                            !this.props.drink.isEmpty && <button id="emptyDrink" className="button" onClick={this.onMakeEmpty}>Empty</button>

                                        )}
                                    </div>
                                </div>
                            </div>
                            <div className="content-container">
                                <h1>{ this.props.drink.title }</h1>
                                <ul>
                                    <li>{ this.props.drink.category }</li>
                                    <li>{ this.props.drink.vol } L</li>
                                    <li>{ this.props.drink.alc } %</li>
                                    <li>{ this.props.drink.price }</li>
                                </ul>
                            </div>
                        </div>
                    ) : (
                        <h1 className="page-header__title">Record with <span>id:{this.props.match.params.id}</span> not found!</h1>
                )}
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        drink: state.drinks.find((drink) => (drink.id === props.match.params.id))
    }
}

const mapDispatchToProps = (dispatch) => ({
        openDrink: (drink) => dispatch(openDrink(drink)),
        emptyDrink: (drink) => dispatch(emptyDrink(drink))
})

export default connect(mapStateToProps, mapDispatchToProps)(DetailedView);