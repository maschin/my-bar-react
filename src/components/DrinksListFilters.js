import React from 'react';
import { connect } from 'react-redux';
import {
    setTextFilter,
    setCategoryFilter
} from '../actions/drinksFilters';


export class DrinksListFilters extends React.Component {
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onCategoryChange = (e) => {
        this.props.setCategoryFilter(e.target.value);
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            type="text"
                            className="text-input"
                            placeholder="Search drinks"
                            value={this.props.drinksFilters.text}
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            value={this.props.drinksFilters.category}
                            onChange={e => this.onCategoryChange(e, 'category')}
                        >
                            <option value=''>-- All --</option>
                            {this.props.categories.map((category) =>
                                <option
                                    key={category}
                                    value={category}>
                                        {category}
                                </option>)}
                        </select>
                    </div>
                </div>
            </div>
        )
    }
}

DrinksListFilters.defaultProps = {
    categories: [
        'Absinthe',
        'Beer',
        'Brandy',
        'Cider',
        'Gin',
        'Horilka',
        'Liqueurs',
        'Mastika',
        'Ouzo',
        'Rakia',
        'Slivovitz',
        'Rum',
        'Sparkling Wine',
        'Tequila',
        'Vodka',
        'Vermouth',
        'Wine',
        'Whisky',
    ]
}

const mapStateToProps = (state) => {
    return {
        drinksFilters: state.drinksFilters
    }
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    setCategoryFilter: (value) => dispatch(setCategoryFilter(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(DrinksListFilters);