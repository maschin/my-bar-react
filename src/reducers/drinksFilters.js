const drinksFiltersReducerDefaultState = {
    text: '',
    category: '',
    sortBy: 'dateBought',
}

export default (state = drinksFiltersReducerDefaultState, action) => {
    switch(action.type) {
        case 'SET_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'SET_CATEGORY_FILTER':
            return {
                ...state,
                category: action.category
            };
        default:
            return state;
    }
}