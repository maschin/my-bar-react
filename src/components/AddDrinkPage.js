import React from 'react';
import { connect } from 'react-redux';

import DrinkForm from '../components/DrinkForm';
import { startAddDrink } from '../actions/drinks';

export class AddDrinkPage extends React.Component {
    onSubmit = (drink) => {
        this.props.startAddDrink(drink);
        this.props.history.push('/');
    };
    render(){
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Bottle</h1>
                    </div>
                </div>
                <div className="content-container">
                    <DrinkForm
                        actionTitle='Add Bottle'
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddDrink: (drink) => dispatch(startAddDrink(drink))
});

export default connect(undefined, mapDispatchToProps)(AddDrinkPage);