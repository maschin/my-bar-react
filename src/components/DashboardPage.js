import React from 'react';
import BarList from '../components/BarList';
import BarSummary from '../components/BarSummary';

export const DashboardPage = () => (
    <div>
        <BarSummary />
        <BarList />
    </div>
);

export default DashboardPage;