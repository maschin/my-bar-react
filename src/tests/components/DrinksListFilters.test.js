import React from 'react';
import { shallow } from 'enzyme';

import { DrinksListFilters } from '../../components/DrinksListFilters';
import { drinksFilters } from '../fixtures/drinksFilters';

let setTextFilter, setCategoryFilter, wrapper;

beforeEach(() => {
   setTextFilter = jest.fn();
   setCategoryFilter = jest.fn();
   wrapper = shallow(
       <DrinksListFilters
           drinksFilters={drinksFilters}
           setTextFilter={setTextFilter}
           setCategoryFilter={setCategoryFilter}
       />
   )
});

test('should render empty DrinksListFilters correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle set text filter', () => {
    const value = 'whisky';
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });

    expect(setTextFilter).toHaveBeenCalledWith(value);
});

test('should handle on category change filter', () => {
   const value = 'beer';
   wrapper.find('select').at(0).simulate('change', {
       target: { value }
   });

   expect(setCategoryFilter).toHaveBeenCalledWith(value);
});
