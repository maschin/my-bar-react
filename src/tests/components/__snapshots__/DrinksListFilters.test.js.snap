// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render empty DrinksListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="Search drinks"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group__item"
    >
      <select
        className="select"
        onChange={[Function]}
      >
        <option
          value=""
        >
          -- All --
        </option>
        <option
          key="Absinthe"
          value="Absinthe"
        >
          Absinthe
        </option>
        <option
          key="Beer"
          value="Beer"
        >
          Beer
        </option>
        <option
          key="Brandy"
          value="Brandy"
        >
          Brandy
        </option>
        <option
          key="Cider"
          value="Cider"
        >
          Cider
        </option>
        <option
          key="Gin"
          value="Gin"
        >
          Gin
        </option>
        <option
          key="Horilka"
          value="Horilka"
        >
          Horilka
        </option>
        <option
          key="Liqueurs"
          value="Liqueurs"
        >
          Liqueurs
        </option>
        <option
          key="Mastika"
          value="Mastika"
        >
          Mastika
        </option>
        <option
          key="Ouzo"
          value="Ouzo"
        >
          Ouzo
        </option>
        <option
          key="Rakia"
          value="Rakia"
        >
          Rakia
        </option>
        <option
          key="Slivovitz"
          value="Slivovitz"
        >
          Slivovitz
        </option>
        <option
          key="Rum"
          value="Rum"
        >
          Rum
        </option>
        <option
          key="Sparkling Wine"
          value="Sparkling Wine"
        >
          Sparkling Wine
        </option>
        <option
          key="Tequila"
          value="Tequila"
        >
          Tequila
        </option>
        <option
          key="Vodka"
          value="Vodka"
        >
          Vodka
        </option>
        <option
          key="Vermouth"
          value="Vermouth"
        >
          Vermouth
        </option>
        <option
          key="Wine"
          value="Wine"
        >
          Wine
        </option>
        <option
          key="Whisky"
          value="Whisky"
        >
          Whisky
        </option>
      </select>
    </div>
  </div>
</div>
`;
